Makefile

```diff
@@ -12,11 +13,22 @@ clean:
 # the "-tags daemon" part is temporary
 .PHONY: test
 test:
-	./scripts/test/unit $(shell go list ./... | grep -v /vendor/)
+	go test -tags daemon -v ${PACKAGES}
 
 .PHONY: test-coverage
 test-coverage:
-	./scripts/test/unit-with-coverage
+	( for pkg in ${PACKAGES}; do \
+		go test -tags daemon \
+			-cover \
+			-coverprofile=profile.out \
+			-covermode=atomic \
+			$$pkg || exit;\
+		\
+		if test -f profile.out; then \
+			cat profile.out >> coverage.txt; \
+			rm profile.out; \
+		fi; \
+	done )
```

Comment:
I'd start by doing
```
PROFILE_FLAGS := -tags daemon
PROFILE_FLAGS += -cover
PROFILE_FLAGS += -coverprofile=profile.out
PROFILE_FLAGS += -covermode=aromic
```
Then using `$(PROFILE_FLAGS)` in the rule (maybe I might leave `-coverprofile=profile.out` inline since that might be clearer). That takes care of a bunch of wrapping by making it obsolete.

I'd deal with the rest of it by pulling the `for` loop out into actual make rules, something along the lines (completely untested):
```
PROFILES := $(patsubst %,%/profile.out,$(PACKAGES))

%/profile.out:
        go test $(PROFILE_FLAGS) -coverprofile=$@ $*

coverage.txt: $(PROFILES)
        cat $^ > $@

.PHONY: test-coverage
test-coverage: coverage.txt
```
I'd stick `profile.out` in `.gitignore` rather than worrying about `rm`.

That assumes that each `$(PACKAGES)` is a valid directory path at the root, maybe it needs a `src/` prefix on the output (so `src/$/profile.out`). Or maybe you prefer to flatten with a `$(patsubst %,%-profile.out,$(subst /,-,$(PACKAGES))` and a corresponding adjustment to the pattern rule (actually I don't recall if `%` in a pattern rule  matches with nested subdirs, so you might be forced down that path).

Maybe `cat $^ > $@` is too opaque for some people, if so `cat $(PROFILES) > coverage.txt` would be completely fine too.